### ISSUE TITLE
Spectralizer causes OBS Crash when it listens to MP4 and I stream the video

### ISSUE THREAD

**Issue by Nass86** [univrsal/spectralizer#58]
Hello,

Thanks for your efforts with Spectralizer. I have a problem I listed on the OBS forum and someone said there was a user reporting similar, earlier this year.

I recorded a video without Spectralizer on Friday - all fine.

On Sunday, I loaded the MP4 into OBS. I turned on Spectralizer, and streamed this to Youtube.

However, when adjusting any setting thereafter, pressing any buttons on OBS - OBS would crash. 

I tried ignoring the computer but it would crash at random, 12 minutes in, 18 minutes in, without me touching the laptop.

The problem happened when Streaming. I managed to later record the whole set with Spectralizer for 2 hours. It crashed when the video ended BUT I did get the MP4 I wanted and have since uploaded to youtube ( https://youtu.be/fUW7qSUzEIk?t=2285 )

Once this had occurred, upon reopening OBS it would instantly crash and remain in an open-crash loop. The only solution was to move the MP4 file so OBS couldn't locate the file - then OBS would launch.

Another point to note was that I was using this with a 150 ms delay set to the audio. Which might be causing the problem or part of it.

OBS Log File: 
https://obsproject.com/logs/BBr4NDpMWisI4TE5

Crash Log: 
https://obsproject.com/forum/threads/obs-crashes-when-playing-mp4-on-a-stream.148861/#:~:text=Attachments-,crash.txt,-69.2%20KB

Hope it can be resolved, I love Spectralizer!

**Comment by univrsal** [comment]
What settings are you using in the spectralizer source?

**Comment by Nass86** [comment]
Thanks for coming back to me so quickly. Here you go!

![A7CF8DDC-C20D-491F-8860-B51E9FB3834F](https://user-images.githubusercontent.com/80380533/136251667-cac1c549-19d4-45cb-a0a3-77823a42ac89.jpeg)

![D7161D48-8CC1-4E34-BBEA-9F2805E32427](https://user-images.githubusercontent.com/80380533/136251693-caf3a10f-6be1-425a-a944-56b6f44e8c32.jpeg)

![8241AC86-AEFF-408F-B243-5FD571AB2E27](https://user-images.githubusercontent.com/80380533/136251724-f357281c-f98d-48af-b2ea-0ae3415b622d.jpeg)

**Comment by Nass86** [comment]
(Nvidia NVENC New)

**Comment by univrsal** [comment]
Yeah those are way too many bins and rounded corners make it even worse. 25 corner points is also way too many.

**Comment by Nass86** [comment]
Thanks. What settings do you suggest? the software indicates no limits /
things to keep an eye on.

>

**Comment by univrsal** [comment]
200 bins is probably as far as I'd go. Rounded corners should be as few as possible, you probably won't even notice the difference in your case so you could just disable them completely. The drawing isn't as efficient as it could be so until I get around to fixing that keeping the complexity of the visualizer low is recommended.

I also assume that that is the cause of the crash. Turning off rounded corners should probably fix it since the rectangular bars are drawn using functions provided by obs which should be safe, whereas the rounded corners are implemented by me.

**Comment by Nass86** [comment]
Thanks. I'll give her another whirl in a week or so and report in.

**Comment by SuslikV** [comment]
Just to clarify, the mentioned "...reporting similar, earlier this year" was this: https://github.com/univrsal/spectralizer/issues/50
